Logging & FPS Tweak

mathieuduperre opened the pull request:
This release includes logging capabilities. a directory (system/logs) gets created and 2 log files (one for the webserver and one for the threads) get created and filled. log level is currently at info, so set of lower if you don't want them to fill up. Setting the log level to debug will create really huge file. log rotation is automatic and the size of each log file can be configured in webApp.py. 

another feature included is a check box to enable/disable FPS Tweak. Brandon implemented a custom FPS feature to measure frame rate when you have multiple ip cameras, preventing a single camera from taking all the available cpu cycle. That created issues with some video feeds as reported by a few people. The checkbox is a flag when a camera is created that will turn on/off this FPS tweak. If you use an iphone with live streamer app, disable the FPS tweak for better result.

BrandonJoffe commented:
Hey @mathieuduperre,

nice work :)

I just have a couple questions.

I noticed you removed a lot of the images from the training set including the unknown dataset. May I ask why?

Is there anything you would like me to add to the readme to help users understand the logging system?

I think it would be a good idea for me to write up some sort of user manual which we can update as we add features/make changes to the system, could you possibly write a few paragraphs describing the logging and how to use it effectively when debugging?

Thanks,
Brandon

mathieuduperre commented:
hi brandon,

oh yeah my apology for removing the pictures. I was debugging the whole learning mechanism and it was taking a long time to learn so i removed all of them and added mine to speed the process (which went from 10 minutes to a few seconds weirdly enough). Anyway my apology and feel free to re-add them and skim mine. i'll put a gitignore in it. 

As for the logging mechanism, you can probably simply add that a directory called logs gets created and that everything get written the log files. It uses logging library so you can increase/decrease verbosity. 

I'm working on updating your upload feature to change it for a learning mechanism, where you learn from the camera feed instead of uploading static pictures. I'll keep you posted when its done and I can add a few things for the user manual.  cheers!
